InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java

```diff
@@ -0,0 +1,90 @@
+package com.ccs.inventorymanagement.route;
+
+import com.ccs.inventorymanagement.config.RouteConfig;
+import com.ccs.inventorymanagement.domain.Clothing;
+import com.ccs.inventorymanagement.domain.Item;
+import com.ccs.inventorymanagement.service.ClothingService;
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.server.HandlerFunction;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+import java.util.function.Function;
+
+public class UpdateClothingHandler implements HandlerFunction<ServerResponse> {
+
+    private final ClothingService service;
+
+    public UpdateClothingHandler(ClothingService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(Request.class)
+                .flatMap(request -> service.update(Clothing.builder()
+                        .id(UUID.fromString(serverRequest.pathVariable(RouteConfig.ID_VARIABLE)))
+                        .name(request.getName())
+                        .description(request.getDescription())
+                        .condition(request.getCondition())
+                        .status(Item.Status.PRESENT)
```

Comment:
What would happen if you attempt to update a clothing item that has been issued because the color was wrong or something? It would mark it as present right? which wouldn't be correct.

Reply:
This is from when I was doing both update and delete in the PUT call, but updated